src/frontend/mame/ui/menu.h

```diff
@@ -64,6 +64,7 @@ class menu
 
 	// append a new item to the end of the menu
 	void item_append(const char *text, const char *subtext, UINT32 flags, void *ref, menu_item_type type = menu_item_type::UNKNOWN);
+	void item_append(const std::string &text, const std::string &subtext, UINT32 flags, void *ref, menu_item_type type = menu_item_type::UNKNOWN);
```

Comment:
Added a move constructor and got rid of the `char *` overload.  Tons of call sites expected nullptr to work, so I've had to do a ton of changes to `""`.  I almost certainly missed some.